release/apparmor.go

```diff
@@ -234,21 +151,38 @@ var (
 	appArmorFeaturesSysPath = "/sys/kernel/security/apparmor/features"
 )
 
-func assessAppArmor() {
+type appArmorProber interface {
+	KernelFeatures() ([]string, error)
+	ParserFeatures() ([]string, error)
+}
+
+type appArmorAssess struct {
+	appArmorProber
+	// level contains the assessment of the "level" of apparmor support.
+	level AppArmorLevelType
+	// summary contains a human readable description of the assessment.
+	summary string
+
+	once sync.Once
+}
+
+func (aaa *appArmorAssess) assess() {
+	aaa.once.Do(func() {
+		aaa.level, aaa.summary = aaa.doAssess()
+	})
+}
+
+func (aaa *appArmorAssess) doAssess() (level AppArmorLevelType, summary string) {
 	// First, quickly check if apparmor is available in the kernel at all.
-	kernelFeatures, err := AppArmorKernelFeatures()
+	kernelFeatures, err := aaa.KernelFeatures()
 	if os.IsNotExist(err) {
-		appArmorLevel = NoAppArmor
-		appArmorSummary = "apparmor not enabled"
-		return
+		return NoAppArmor, "apparmor not enabled"
```

Comment:
This is nice, after writing this I was thinking about changing the code around to avoid the risk of forgetting to return. This makes the problem moot.